src/mon/OSDMonitor.cc

```diff
@@ -14264,6 +14286,27 @@ bool OSDMonitor::prepare_pool_op(MonOpRequestRef op)
     }
     break;
 
+
+  case POOL_OP_DELETE_SNAP_AGAIN:
+    {
+        // Optimally we could re-use get_purged_snaps. However, the purged snaps are pruned.
+        // removed_snaps is also cleared.
+        //
+        // Note: for this to work with selfmanaged snaps IoCtx::snap_list should include
+        //       self-managed snaps. Currently the iteration is up to last snapid (pool snaps)
+        for (uint64_t i = 1; i < pp.get_snap_seq() ; i++) {
```

Comment:
is u64 the correct choice here? how far can snap_seq reach?
I'd use a signed integer, so that - if get_snap_seq) would ever be modified to return an error
as a negative number, we won't crash here

Reply:
get_snap_seq() is a const getter for:
```
snapid_t snap_seq = 0;        ///< seq for per-pool snapshot
```
 > is u64 the correct choice here? how far can snap_seq reach?

```
struct snapid_t {
  uint64_t val;
   ..
}
```